modules/internal/ChapelIO.chpl

```diff
@@ -465,7 +465,7 @@ module ChapelIO {
         if st == QIO_AGGREGATE_FORMAT_JSON {
           start = new ioLiteral("{");
         } else if st == QIO_AGGREGATE_FORMAT_CHPL {
-          start = new ioLiteral("new " + typeToString(t) + "(");
+          start = new ioLiteral("new " + typeToString(t).c_str() + "(");
```

Comment:
This seems like this should go into the list of "changes a developer should expect to understand" list that comes with this merge then.  

(Personally, I think it's regrettable that we require a method rather than a cast for this conversion, which makes it seem like a big outlier.  From your comment, I take it that this is more of a "current compiler architecture and complexity" issue than a "language tractability" issue, is that right?  I.e., if we had dozens of people working on a v2, there's no reason to believe that one couldn't support conversions from strings to c_strings via casts?)

Would having conversions be defined via module code rather than special cases in resolution as @mppf has started working on potentially help this case?